functions/src/modules/TE.functions.ts

```diff
@@ -67,19 +74,166 @@ export const processAllotment = functions.https.onCall(
     const coordinator = functions.config().coordinator;
 
     // Notify the assignee
-    await admin.database().ref(`/email/notifications`).push({
-      template: 'track-editing-allotment',
-      to: assignee.emailAddress,
-      bcc: coordinator.email_address,
-      params: {
-        tasks: tasksForEmail,
-        assignee: assignee,
-        comment: comment,
-        date: moment()
-          .utcOffset(coordinator.utc_offset)
-          .format('DD.MM'),
-        uploadURL: `${functions.config().website.base_url}/te/upload/`,
-      },
-    });
+    await admin
+      .database()
+      .ref(`/email/notifications`)
+      .push({
+        template: 'track-editing-allotment',
+        to: assignee.emailAddress,
+        bcc: coordinator.email_address,
+        params: {
+          tasks: tasksForEmail,
+          assignee: assignee,
+          comment: comment,
+          date: moment()
+            .utcOffset(coordinator.utc_offset)
+            .format('DD.MM'),
+          uploadURL: `${functions.config().website.base_url}/te/upload/`,
+        },
+      });
   }
 );
+
+const teUploadsBucket = `https://te.uploads.${
+  functions.config().storage['root-domain']
+}`;
+
+const editedBucket = `https://edited.${
+  functions.config().storage['root-domain']
+}`;
+
+export const handleTeSubmission = functions.storage
+  .bucket(teUploadsBucket)
+  .object()
+  .onFinalize(async object => {
+    const filePath = object.name;
+
+    const directories = path.dirname(filePath).split('/');
+    if (directories.length !== 2) {
+      throw new Error(
+        `File "${filePath}" is not uploaded to the appropriate folder`
+      );
+    }
+
+    const teUser = admin.auth().getUser(directories[0]);
```

Comment:
Here we need not check this. Users are allowed to write to their own folders only by storage security rules.

You'd rather get the user id from `context.auth.uid` (see [doc](https://firebase.google.com/docs/reference/functions/functions.EventContext#.auth)).